README.md

```diff
@@ -33,16 +37,56 @@ const layoutEngine = constructLayoutEngine({routes, applications})
 start()
 ```
 
-
 ### Constructing Routes
+
 single-spa-layout supports two different sytanxes defining applications and routes at this time:
+
 - [JSX](#JSX) (compiles to JSON syntax)
 - [JSON](#JSON)
 
+#### Route Object types
+
+1. [Router](#Router)
+2. [Application](#Application)
+3. [Route](#Route)
+
+##### Router
+
+The root object/node. Router can have the following options:
+
+- mode: 'history' | 'hash'
+- base: base url where the router is active
+- containerEl: which cssSelector is where the routing should take place (inferred in HTML syntax)
+- routes: Array of routes or Application objects. _required_
```

Comment:
I'm not in love with _required_ at the end but I wasn't sure if using JSDOC style syntax here was a good idea or not.

Reply:
I'm renaming this to `children` in #21